### PR TITLE
feat(materials): Helm Chart material support

### PR DESCRIPTION
**Pull request by Javirln**
Add support for Helm Charts files as material types. These get validated before they are uploaded to the CAS and attached to the attestation process.

The validations checks for the following:
- Incoming file is a compressed tarball.
- `Charts.yaml` and `values.yaml` files are present on the uncompressed directory and it's a valid YAML file.

Closes https://github.com/chainloop-dev/chainloop/issues/684

### CODE REVIEW

**Comment by migmartri** [comment]
Awesome @Javirln 

As you can see, the CI is complaining that the DCO is not completed. This means that you are not committing with the signed-off line `git commit -s ...`

I'd suggest that you squash all your commits and amend it to have the signed off. Happy to help if you have any questions

Thanks

File: README.md [hunk 1]
```diff
@@ -130,6 +130,7 @@ Chainloop supports the collection of the following pieces of evidence types:
 - [OpenVEX](https://github.com/openvex)
 - [SARIF](https://docs.oasis-open.org/sarif/sarif/v2.1.0/)
 - [JUnit](https://www.ibm.com/docs/en/developer-for-zos/14.1?topic=formats-junit-xml-format)
+- [Helm Charts](https://helm.sh/docs/topics/charts/)
```
Comment (migmartri) [inline README.md:133]: I'd keep it singular for consistency

```suggestion
- [Helm Chart](https://helm.sh/docs/topics/charts/)
```

File: internal/attestation/crafter/materials/helmchart.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	uncompressedStream, err := gzip.NewReader(f)
+	if err != nil {
+		return nil, fmt.Errorf("can't uncompress file, unexpected material type: %w", err)
+	}
+
+	// Create a tar reader
+	tarReader := tar.NewReader(uncompressedStream)
+
+	// Flags to track whether required files are found
+	chartFileValid, chartValuesValid := false, false
```
Comment (migmartri) [inline internal/attestation/crafter/materials/helmchart.go:74]: nitpick and curious to knwo your take. I think you can technically leverage go default values instead.

```
var chartFileValid, chartValuesValid bool
```

File: internal/attestation/crafter/materials/helmchart.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+	// Iterate through the files in the tar archive
+	for {
```
Comment (migmartri) [inline internal/attestation/crafter/materials/helmchart.go:77]: does this work with an empty tarball?

File: README.md (hunk 1 above)
Comment (migmartri) [inline README.md:133]: also, if you want to add the documentation in this PR. I am ok if you do it in another one, you'll need to also update https://github.com/chainloop-dev/chainloop/blob/main/docs/docs/reference/operator/contract.mdx

That's the source code of the doc site. So probably let's do it in another PR

**Review by migmartri — APPROVED** [review]

File: internal/attestation/crafter/materials/helmchart.go (hunk 2 above)
Comment (jiparis) [reply internal/attestation/crafter/materials/helmchart.go:74]: +1

File: internal/attestation/crafter/materials/helmchart.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	api "github.com/chainloop-dev/chainloop/internal/attestation/crafter/api/attestation/v1"
+	"github.com/chainloop-dev/chainloop/internal/casclient"
+	"github.com/rs/zerolog"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	ChartFileName      = "Chart.yaml"
+	ValuesYamlFileName = "values.yaml"
```
Comment (jiparis) [inline internal/attestation/crafter/materials/helmchart.go:36]: Since these are in a shared package, I would suggest to prefix both constants with an agreed pattern to avoid conflicts. Something like `ChartFileName` and `ChartValuesFileName`.

Comment (migmartri) [reply internal/attestation/crafter/materials/helmchart.go:36]: true, and in fact they don't need to be exposed right?

File: internal/attestation/crafter/materials/helmchart.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	if !chartFileValid || !chartValuesValid {
+		return nil, fmt.Errorf("missing required files in the helm chart: Chart.yaml and values.yaml")
+	}
+
+	// Upload and craft the chart
+	return uploadAndCraft(ctx, c.input, c.backend, filepath, c.logger)
+}
+
+// validateYamlFile validates the YAML file just by trying to unmarshal it
+func validateYamlFile(r io.Reader) error {
```
Comment (jiparis) [inline internal/attestation/crafter/materials/helmchart.go:121]: Please add a receiver to this method so that it's not exposed at package level (unless we want to expose it for some reason).

Comment (Javirln) [reply internal/attestation/crafter/materials/helmchart.go]: Hmm no no, we don't need to expose it. Eventually in the future this can belong to a shared/utils part to be reused in the the code. Right now, there is no need for that neither for it be exposed.

File: internal/attestation/crafter/materials/helmchart.go (hunk 4 above)
Comment (Javirln) [reply internal/attestation/crafter/materials/helmchart.go]: Added a common pattern and made them private outside the package.

**Comment by migmartri** [comment]
Looks good to me! Feel free to merge it!
